src/fetch/payouts.py

```diff
@@ -420,15 +448,57 @@ def construct_payouts(
     performance_reward = complete_payout_df["primary_reward_cow"].sum()
     participation_reward = complete_payout_df["secondary_reward_cow"].sum()
     quote_reward = complete_payout_df["quote_reward_cow"].sum()
-    protocol_fee = complete_payout_df["protocol_fee_eth"].sum()
+    raw_protocol_fee_wei = int(complete_payout_df.protocol_fee_eth.sum())
+
+    # We now decompose the raw_protocol_fee_wei amount integrator fees
+    # into actual protocol fee and integrator fees. For convenience,
+    # we use a dictionary integrators_fees_wei that contains the the
+    # destination address of an integrator as a key, and the value is the
+    # amount in wei to be transferred to that address, stored as an int.
+
+    integrators_fees_wei: dict[str, int] = {}
+    for _, row in integrators_fees_df.iterrows():
+        if row["integrators_list"] is None:
+            continue
+
+        # We assume the two lists used below, i.e.,
+        # integrators_list and integrators_payments_in_eth,
+        # are "aligned".
```

Comment:
A [mailing list post](https://www.postgresql.org/message-id/flat/592DF768-0FF1-4795-AAE9-B5B684A0469D%40gmail.com) mentions this:

>  I don't remember exactly
where, but I'm sure it's still described somewhere.  So it is documented
behavior that an aggregate without its own ORDER BY will see the rows in
whatever order the FROM clause supplies them.
I'm not very keen on recommending that the OP insert an ORDER BY into
each aggregate call, because that would cause a separate sort for each
aggregate (unless someone's improved that recently while I wasn't
looking).
regards, tom lane

So our query should always return results satisfying the assumption.